### DISCUSSION TITLE
insert data from a trusted server

### DISCUSSION THREAD

**Discussion by andykais** [supabase/supabase#335] [Questions] [answered]
Hi, I am just starting out with supabase. I want to insert data into postgres from a trusted server periodically. I am using the supabase-js client. I gave it my `service_role` `secret` api key and what I imagined was that this would let me insert anything into any of my tables. Instead, performing an insert 
```js
const { data, error } = await supabase.from('movies').insert([row_data])
```
results in this error:
```js
{
  data: null,
  error: {
    hint: null,
    details: null,
    code: '42P01',
    message: 'relation "public.movies" does not exist'
  }
}
```
The table does exist, and all I did was copy the commands given to me on this page `/api/default?resource=movie`. What am I missing to get this off the ground? On a related note, I do not want users inserting into this table, only those with a private server key should be able to.

**Comment by steve-chavez** [comment]
Hey @andykais,

On what schema did you create your `movies` table?

If you chose a schema different than `public`(the default for supabase-js), you'd have to declare it like:

```js
const supabase = createClient(URL, KEY, {schema: OTHER});
```

**Reply by andykais** [reply]
I believe its public

![screenshot(270)](https://user-images.githubusercontent.com/6627147/101261994-31858780-3709-11eb-907f-72ab11a49544.png)

**Comment by andykais** [comment, marked as answer]
oh good lord that was dumb. `movies` should have been `movie`. Crisis averted. Thanks for the help anyways :sweat_smile:
